src/types.jl

```diff
@@ -24,7 +24,7 @@ mutable struct FirstOrderModel{T, S, F, G} <: AbstractNLPModel{T, S}
     x::S;
     name::AbstractString = "first-order model",
   ) where {T, S, F <: Function, G <: Function}
-    meta = NLPModelMeta(length(x), x0 = x, name = name)
+    meta = NLPModelMeta(length(x), x0 = x, name = name, lvar = zeros(length(x)), uvar = zeros(length(x)) .= Inf)
```

Comment:
If you let the user pass keyword args to the `FirstOrderModel` constructor, you can forward them to that of `meta`,